src/components/cell-control/textarea.jsx

```diff
@@ -0,0 +1,117 @@
+import PropTypes from 'prop-types';
+import React, { forwardRef, useImperativeHandle, useRef, useState } from 'react';
+import cautionSvg from '../../svgs/caution.svg';
+import CellControl from './cell-control.jsx';
+import Icon from '../icon/icon.jsx';
+import Tooltip from '../tooltip/tooltip.jsx';
+import styles from './textarea.css';
+import useValidation from '../../hooks/use-validation.jsx';
+
+const Textarea = forwardRef(function Textarea(props, ref) {
+  const ids = {
+    invalidIcon: `${props.id}-invalid-icon`,
+    invalidTooltip: `${props.id}-invalid-tooltip`,
+  };
+  const refs = {
```

Comment:
Does this need to be a dictionary, since it has only one entry?

Reply:
🤷  No but I see it as a useful pattern for MDS components. We have between 0, 1, or many refs. Setting up the structure now makes it easier for the next dev? Only time will tell! ⏳